staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go

```diff
@@ -49,21 +50,64 @@ type Controller struct {
 
 	queue workqueue.RateLimitingInterface
 
-	staticSpec     *spec.Swagger
+	staticSpec *spec.Swagger
+
 	openAPIService *handler.OpenAPIService
 
-	// specs per version and per CRD name
-	lock     sync.Mutex
-	crdSpecs map[string]map[string]*spec.Swagger
+	// specs by name. The specs are lazily constructed on request.
+	lock        sync.Mutex
```

Comment:
nit: it's not obvious whether the lock is for the map only or also the values (pointed to).

Reply:
Added some comments for both the lock and the underlying map struct. Let me know if they're still not clear.